### PR TITLE
Copy sprint2 ko unit test

### PR DESCRIPTION
**Pull request by kkaungyyang**

### CODE REVIEW

File: web-component/inputRT.js [hunk 1]
```diff
@@ -97,6 +349,27 @@
...
 
+        /**
+         * Internal function to determine the correct element in concern. 
+         * @param {*} mode 
+         * @returns {string} an html element in concern
+         */
+        _choose_element(mode) {
+            let retVal = "";
+            switch (mode) {
+                case "custom":
```
Comment (ericsengithub) [inline web-component/inputRT.js]: please add the comment about custom

File: web-component/inputRT.js [hunk 2]
```diff
@@ -63,6 +80,241 @@
...
+         */
+        set url(val) {
+            if (val !== '') {
+                this.setAttribute('url', val);
+            } else {
+                this.removeAttribute('url');
+            }
+        }
+
+        _init_bootstrap_URL(){
```
Comment (ericsengithub) [inline web-component/inputRT.js]: please add a comment here to describe what it is doing

File: web-component/inputRT.js [hunk 3]
```diff
@@ -63,6 +80,241 @@
             }
         }
 
+        /**
```
Comment (ericsengithub) [inline web-component/inputRT.js]: Please order this in ABC order in order of getter -> setter -> init -> unint  -> register -> unregister

File: web-component/inputRT.js [hunk 4]
```diff
@@ -97,6 +349,27 @@
...
```
Comment (ericsengithub) [inline web-component/inputRT.js]: Can we move this under constructor, connected callback, attribute changed callback

File: web-component/inputRT.js [hunk 5]
```diff
@@ -112,9 +385,16 @@
             this._messengerSlot = this.shadowRoot.querySelector('slot[name=messenger]');
             this._textSlot = this.shadowRoot.querySelector('slot[name=text]');
             this._appendSlot = this.shadowRoot.querySelector('slot[name=append]');
+            this._linkSlot = this.shadowRoot.querySelector('slot[name=link]');
 
             // Initialize attributes
             this._init_mode();
```
Comment (ericsengithub) [inline web-component/inputRT.js]: Todo for me (Eric), comment on the order about how things are initialized

File: web-component/inputRT.js [hunk 6]
```diff
@@ -63,6 +80,241 @@
             }
         }
 
+        /**
+         * get bootstrap()
```
Comment (ericsengithub) [inline web-component/inputRT.js]: Please make the spacing across the file unique, as in one newline after every function.

File: web-component/inputRT.js [hunk 7]
```diff
@@ -1,9 +1,25 @@
+//function allows keycode and template to only be accessed by inputRT
 (function(){
     // Dictionary for comparing key presses
     const KEYCODE = {
         ENTER: 13
     };
 
+    const SIZES = {
```
Comment (ericsengithub) [inline web-component/inputRT.js]: Please add a comment describing what this is for

File: web-component/inputRT.js [hunk 8]
```diff
@@ -63,6 +80,241 @@
...
+        set disabled(val) {
+            const isDisabled = Boolean(val);
+            if (isDisabled) {
+                this.setAttribute('disabled', val);
+            } else {
+                this.removeAttribute('disabled');
+            }
+        }
+
+        _init_disabled(){
```
Comment (ericsengithub) [inline web-component/inputRT.js]: Please comment on what this function is doing

File: web-component/inputRT.js [hunk 9]
```diff
@@ -63,6 +80,241 @@
...
+        set width(val) {
+            const isWidth = String(val);
+            if (isWidth) {
+                this.setAttribute('width', val);
+            } else {
+                this.removeAttribute('width');
+            }
+        }
+
+        _init_width(){
```
Comment (ericsengithub) [inline web-component/inputRT.js]: Please add comment about what this is doing

File: web-component/inputRT.js (hunk 7 above)
Comment (kkaungyyang) [reply web-component/inputRT.js]: done

File: web-component/inputRT.js (hunk 6 above)
Comment (kkaungyyang) [reply web-component/inputRT.js]: done.

File: web-component/inputRT.js (hunk 2 above)
Comment (kkaungyyang) [reply web-component/inputRT.js]: done

File: web-component/inputRT.js (hunk 9 above)
Comment (kkaungyyang) [reply web-component/inputRT.js]: done.

File: web-component/inputRT.js (hunk 8 above)
Comment (kkaungyyang) [reply web-component/inputRT.js]: done

File: web-component/inputRT.js (hunk 4 above)
Comment (kkaungyyang) [reply web-component/inputRT.js]: done.

File: web-component/inputRT.js (hunk 1 above)
Comment (kkaungyyang) [reply web-component/inputRT.js]: done.

**Pull request merged**